Remove merge editor settings from the Settings UI

The merge editor is not used in Code Editor. This removes the Merge Editor
group from the Features section of the Settings UI.

Index: b/src/vs/workbench/contrib/preferences/browser/settingsLayout.ts
===================================================================
--- a/src/vs/workbench/contrib/preferences/browser/settingsLayout.ts
+++ b/src/vs/workbench/contrib/preferences/browser/settingsLayout.ts
@@ -365,11 +365,6 @@ export const tocData: ITOCEntry<string>
 					settings: ['notebook.*', 'interactiveWindow.*']
 				},
 				{
-					id: 'features/mergeEditor',
-					label: localize('mergeEditor', 'Merge Editor'),
-					settings: ['mergeEditor.*']
-				},
-				{
 					id: 'features/issueReporter',
 					label: localize('issueReporter', 'Issue Reporter'),
 					settings: ['issueReporter.*'],
